components/camera/videosource/webcam.go

```diff
@@ -193,54 +194,141 @@ func makeConstraints(attrs *WebcamAttrs, debug bool, logger golog.Logger) mediad
 	}
 }
 
-// findCamera finds a video device and returns a camera with that video device as the source.
-func findCamera(ctx context.Context, attrs *WebcamAttrs, logger golog.Logger) (camera.Camera, error) {
+// findCamera finds a video device and returns a reconfigurable camera with that video device as the source.
+func findCamera(
+	ctx context.Context,
+	attrs *WebcamAttrs,
+	label string,
+	logger golog.Logger,
+) (resource.Reconfigurable, error) {
+	var cam camera.Camera
+	var err error
+
 	debug := attrs.Debug
 	constraints := makeConstraints(attrs, debug, logger)
-	if attrs.Path != "" {
-		return tryWebcamOpen(ctx, attrs, attrs.Path, false, constraints, logger)
+	if label != "" {
+		cam, err = tryWebcamOpen(ctx, attrs, label, false, constraints, logger)
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot open webcam")
+		}
+	} else {
+		source, err := gostream.GetAnyVideoSource(constraints, logger)
+		if err != nil {
+			return nil, errors.Wrap(err, "found no webcams")
+		}
+		cam, err = makeCameraFromSource(ctx, source, attrs)
+		if err != nil {
+			return nil, errors.Wrap(err, "cannot make webcam from source")
+		}
 	}
 
-	source, err := gostream.GetAnyVideoSource(constraints, logger)
+	return camera.WrapWithReconfigurable(cam, camera.Named(model))
+}
+
+// getLabelFromCameraOrPath returns the path from the camera or an empty string if a path is not found.
+func getLabelFromCamera(cam camera.Camera, logger golog.Logger) string {
+	src, err := camera.SourceFromCamera(cam)
 	if err != nil {
-		return nil, errors.Wrap(err, "found no webcams")
+		logger.Errorw("cannot get source from camera", "error", err)
+		return ""
 	}
-	return makeCameraFromSource(ctx, source, attrs)
+
+	info, err := gostream.InfoFromMediaSource[image.Image, prop.Video](src)
+	if err != nil {
+		logger.Errorw("could not get info from media source", "error", err)
+		return ""
+	}
+
+	labelParts := strings.Split(info.Label, mediadevicescamera.LabelSeparator)
```

Comment:
I still think this is a leaky abstraction and we should update at least the gostream PR to give you the labels already split out. I'd call this `LabelsFromMediaSource` to limit what we're exposing in response to what we're using.

Reply:
Done